frontend/model/notifications/periodicNotifications/index.js

```diff
@@ -0,0 +1,77 @@
+'use strict'
+
+import sbp from '@sbp/sbp'
+import { Vue } from '@common/common.js'
+// $FlowFixMe
+import { objectOf, string, isFunction } from '@model/contracts/misc/flowTyper.js'
+import { MINS_MILLIS } from '@model/contracts/shared/time.js'
+import { PERIODIC_NOTIFICATION_TYPE } from './constants.js'
+
+const { MIN1, MIN5, MIN30 } = PERIODIC_NOTIFICATION_TYPE
+
+const every1MinTimeout = { queue: [], state: {}, delay: 0.2 * MINS_MILLIS }
+const every5MinTimeout = { queue: [], state: {}, delay: 5 * MINS_MILLIS }
+const every30MinTimeout = { queue: [], state: {}, delay: 30 * MINS_MILLIS }
+
+const typeToObjectMap = {
+  [MIN1]: every1MinTimeout,
+  [MIN5]: every5MinTimeout,
+  [MIN30]: every30MinTimeout
+}
+
+const validateNotificationData = objectOf({
+  stateKey: string,
+  emitCondition: isFunction,
+  emit: isFunction,
+  shouldClearStateKey: isFunction
+})
+
+async function runQueueRecursive (data) {
+  const rootState = sbp('state/vuex/state')
+  const rootGetters = sbp('state/vuex/getters')
+  const firedMap = rootState.periodicNotificationAlreadyFiredMap
+  const callWithStates = func => func.call(data.state, { rootState, rootGetters })
```

Comment:
Calling `emitCondition()`, `emit()`, and `shouldClearStateKey()` this way was kind of inspired from the way we use the reference the contract states & getters inside the `process()`, `sideEffect()` functions. I wanted to be able to reference the vuex state and getters inside them like an example below

```js
emit ({ rootState, rootGetters }) {
  const { currentGroupId } = rootState
  const { ourUsername, ourPayments } = rootGetters
  ...
}
```